src/services/llm/config.py

```
@@ -65,6 +65,49 @@ def _get_llm_mode_str() -> str:
+
+    provider_type: Literal["api", "local"] = "local" if is_local_llm_server(base_url) else "api"
+
+    # Check if API key is required (not required for local providers)
+    requires_key = (
+        provider_type == "api" or os.getenv("LLM_API_KEY_REQUIRED", "false").lower() == "true"
+    )
+
+    if requires_key and not api_key:
+        raise LLMConfigError(
+            "LLM_API_KEY not set, please configure it in .env file or activate a provider"
+        )
```

Ruff (0.14.11)

Avoid specifying long messages outside the exception class

(TRY003)